.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,19 @@
+<!--
+  ~ Housekeeping Checklist ~
```

Comment:
I am assuming we will be uncommenting this MD file before going to prod, right?
Actually, when previewing (uncommented) this file, the formatting seems a little bit off compared to other markdowns.
For reference, please check https://github.com/github/docs/blob/main/CONTRIBUTING.md and its corresponding code.

<img width="1360" alt="image" src="https://user-images.githubusercontent.com/7558499/204730537-7ce9ae26-57b7-4ef3-a8ab-0c888fd0f633.png">

Reply:
No, this file will not be uncommented before merging this PR.

We currently don't have a CONTRIBUTING.md but will probably want to include a checklist like this in the guide so I put the content there for now.

We need a separate PR to work on the CONTRIBUTING.md

Reply:
The goal wasn't really to have a CONTRIBUTING.md file in this PR, I just felt like this checklist didn't belong in PR template, so it was moved there.
But if everyone feels like it make more sense to have it in the PR template, we can still change it, I don't have strong opinion here

Reply:
i'd love to have a CONTRIBUTING.md and a CODE_OF_CONDUCT.md
we should give more thoughts to it and have separate PRs

I quite like the CONTRIBUTING.md from vue
they also have a Pull Request Guidelines section
https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md